flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamConfig.java

```diff
@@ -508,12 +515,18 @@ public void setStateBackend(StateBackend backend) {
 		if (backend != null) {
 			try {
 				InstantiationUtil.writeObjectToConfig(backend, this.config, STATE_BACKEND);
+				this.config.setBoolean(STATE_BACKEND_USE_MANAGED_MEMORY, backend.useManagedMemory());
```

Comment:
For the purpose to test method, I think we should use `setStateBackendUsesManagedMemory(backend.useManagedMemory());` here